src/Pages/Article.js

```diff
@@ -0,0 +1,120 @@
+import React, { useState } from 'react';
+import { useHistory, Link } from 'react-router-dom';
+import Footer from '../Components/Footer';
+import NavBar from '../Components/NavBar';
+import { useScrollToTop } from '../Helpers/helpers';
+import { getPostById as getPost, doesPostExist as isPost } from '../posts'; 
+import Comments from '../Components/Comments';
+import { persistUserAction as registerVote, getCurrentInteractionOnPost as getVote } from '../Helpers/storageHelper';
+
+function Article(props) {
+  const [vote, setVote] = useState(getVote(props.location.state.id));
+  const history = useHistory();
+  useScrollToTop();
+
+  let path = props.location.pathname.split("/");
+  let id = null;
+
+  try {
+    id = props.location.state.id;
+  } catch (idNotFound) {
+    console.log("could not find any id @ state.id \n trying the title....");
+    try {
+      id = props.location.state.postTitle 
+    } catch (titleNotFound) {
+      console.log("could not find any title @ state.postTitle\ntrying to extract from the url");
+      try {
+        id = path[path.length - 1].replaceAll("-", " ")
+      } catch (urlTitleNotFound) {
+        console.log("No such post exists");
+      }
+    }
+  }
+
+  let post  = null;
+
+  if(isPost(id)) {
+    console.log("found id: " + id)
+    post = getPost(id);
+    console.log(post);
+  } else {
+    console.log("could not find an id");
+    return (
+      <>
+        <NavBar />
+        <main role="main" className="container" style={{height: "100vh", paddingTop: "160px"}}>
```

Comment:
You have to be careful when using the vh notation for your styling...
nothing to worry about in this instance but just to draw your attention to it...probably read a bit on it Sir